docs/content/contributing/terraform/terraform-contribution-flow/_index.md

```diff
@@ -0,0 +1,313 @@
+
+For consistent contributors but also Azure-org members in general it is possible to get invited as collaborator of the module repository which enables you to work on branches instead of forks. To get invited get in touch with the module owner since it's the module owner's decision who gets invited as collaborator.
+
+{{< /hint >}}
+
+<br>
+
+---
+
+<br>
+
+### 2. Prepare your Azure test environment
+
+AVM performs end-to-end (e2e) test dpeloyments of all modules in Azure for validation. We recommend you to perform a local e2e test deployment of your module before you create a PR to the upstream repository. Especially because the e2e test deployment will be triggered automatically once you create a PR to the upstream repository.
+
+1. Have/create an Azure Active Directory Service Principal with at least `Contributor` & `User Access Administrator` permissions on the Management-Group/Subscription you want to test the modules in. You might find the following links useful:
+- [Create a service principal (Azure CLI)](https://learn.microsoft.com/en-us/cli/azure/azure-cli-sp-tutorial-1) - _**Recommended**_
+- [Create a service principal (Azure Portal)](https://learn.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal)
+- [Create a service principal (PowerShell)](https://learn.microsoft.com/azure/active-directory/develop/howto-authenticate-service-principal-powershell)
+- [Find Service Principal object ID](https://learn.microsoft.com/azure/cost-management-billing/manage/assign-roles-azure-service-principals#find-your-spn-and-tenant-id)
+- [Find managed Identity Service Principal](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-view-managed-identity-service-principal-portal)
+- Note down the following pieces of information
+- Application (Client) ID
+- Service Principal Secret (password)
+- **Optional:** Tenant ID
+- **Optional:**  Subscription ID
+
+```bash
+# Linux/MacOs
+export ARM_SUBSCRIPTION_ID=$(az account show --query id --output tsv) # or set <subscription_id>
+export ARM_TENANT_ID=$(az account show --query tenantId --output tsv) # or set <tenant_id>
+export ARM_CLIENT_ID=<client_id>
+export ARM_CLIENT_SECRET=<service_principal_password>
+
+# Windows/Powershell
+$env:ARM_SUBSCRIPTION_ID = $(az account show --query id --output tsv) # or set <subscription_id>
+$env:ARM_TENANT_ID = $(az account show --query tenantId --output tsv) # or set <tenant_id>
+$env:ARM_CLIENT_ID = "<client_id>"
+$env:ARM_CLIENT_SECRET = "<service_principal_password>"
+
+```
+
+2. Change to the root of your module repository and run `./avm docscheck` (Linux/MacOs) / `avm.bat docscheck` (Windows) to verify the container image is working as expected or needs to be pulled first. You will need this later.
+
+<img src="/Azure-Verified-Modules/img/contribution/pullImage.png" alt="Pull latest azterraform container image." width=100%>
+
+<!--
+1. Create an UAMI in your Azure test subscription.
```

Comment:
We don't need UMIs as they are not supported for client secrets

Reply:
leftover commment, removed